### PR TITLE
TE-2008 Add lazy loading

### PR DESCRIPTION
**Pull request by Jonnygwi**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-2008)

### What **one** thing does this PR do?
Proposal for lazy loading

### Any other notes
- The debounce was increased to make the lazy loading more clear

Kapture 2019-03-15 at 15 39 45

### CODE REVIEW

File: src/components/layout/LazyLoader/component.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import React, { PureComponent } from 'react';
+import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+
+import { isVisible } from './utils/isVisible';
```
Comment (marcdavi-es) [inline src/components/layout/LazyLoader/component.js]:
```suggestion
import { getIsVisible } from './utils/getIsVisible';
```

File: src/components/layout/LazyLoader/component.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            ...componentProps,
+          }
+        : componentProps;
+
+    const componentToRender =
+      lazyProps !== undefined || this.state.shouldRender
+        ? React.cloneElement(lazyComponent, componentToRenderProps)
+        : null;
+
+    return <div ref={this.createComponentRef}>{componentToRender}</div>;
```
Comment (marcdavi-es) [inline src/components/layout/LazyLoader/component.js]: I wonder if this `div` might break some component structures?

Can we just use a dummy element which does not wrap `componentToRender`? Then maybe we can render the dummy first, use to watch the position relative to the viewport, then when it is visible, swap it out for `componentToRender` ?

Comment (Jonnygwi) [reply src/components/layout/LazyLoader/component.js]: Hmmmmmm Mmmmmmm.

Comment (Jonnygwi) [reply src/components/layout/LazyLoader/component.js]: Good thought. I'll give it a try.

Comment (Jonnygwi) [reply src/components/layout/LazyLoader/component.js]: I'm going to leave it as a div for now and if it effects some component layouts then we can make optimisations then.. As for the moment it's only going to be used in Images and Gallery so it should change anything there.

**Pull request closed without merging**